### PR TITLE
WINC-956: Reduce direct use of the AWS API in e2e tests

### PR DESCRIPTION
**Pull request by sebsoto**
This PR brings the AWS provider in line with the rest of the providers, by using an existing worker MachineSet as a base for the Windows MachineSet created in our e2e tests. This removes complexity related to dealing with the AWS CLI, simplifying the provider code, and reducing the potential of test failure due to AWS API rate limiting issues that can infrequently occur.
    
The only remaining API call is DescribeImages, which is required as the use of the MachineAPI for AWS requires a specific imageID be provided.

### CODE REVIEW

**Comment by mtnbikenc** [comment]
@sebsoto Thanks for working on this.  It looks like throttling on AWS disproportionately affects WMCO PRs.

File: test/e2e/providers/aws/aws.go [hunk 1]
```diff
@@ -140,185 +106,30 @@ func getLatestWindowsAMI(ec2Client *ec2.EC2) (string, error) {
...
+		return nil, fmt.Errorf("found 0 worker role machines")
 	}
-
-	sgID, err := a.getClusterWorkerSGID()
-	if err != nil {
-		return nil, fmt.Errorf("unable to get security group id: %v", err)
-	}
-
-	subnet, err := a.getSubnet()
+	foundSpec := &mapi.AWSMachineProviderConfig{}
```
Comment (aravindhp) [inline test/e2e/providers/aws/aws.go:119]: I suggest naming this `linuxWorkerSpec`.

**Review by aravindhp — APPROVED** [review]
Thanks for doing this, @sebsoto

**Comment by alinaryan** [comment]
Thanks for working on this, 
/lgtm

**Comment by aravindhp** [comment]
/retest-required

```
Timed out waiting for the catalog source oo-zdxqb to become ready after 10 minutes.
```

**Pull request merged**